src/setup-beam.js

```diff
@@ -560,38 +559,50 @@ function getRunnerOSVersion() {
   return containerFromEnvImageOS
 }
 
-async function get(url0, pageIdxs) {
-  async function getPage(pageIdx) {
-    const url = new URL(url0)
-    const headers = {}
-    const GithubToken = getInput('github-token', false)
-    if (GithubToken && url.host === 'api.github.com') {
-      headers.authorization = `Bearer ${GithubToken}`
-    }
+async function getUrlResponse(url, headers, attempt = 1) {
+  try {
+    const response = await fetch(url, { headers })
+    const contentType = response.headers.get('content-type') || ''
 
-    if (pageIdx !== null) {
-      url.searchParams.append('page', pageIdx)
+    if (!response.ok) {
+      throw new Error(`Got ${response.statusCode} from ${url}`)
     }
```

Comment:
Just checking if the response is ok or not. We can get more detailed with status code checking if we need, but I don't see a reason.

Reply:
If we weren't getting more detail no more detail is required, at this moment. There's been requests to include more detail around errors, but that's for a different pull request.